### PR TITLE
Additional options for low-level profiling

### PR DESCRIPTION
**Pull request by olorin**
/cc @jystic

### CODE REVIEW

**Comment by jacobstanley** [comment]
@olorin nice

Would there be a use for unstripped executables without profiling enabled? might be handy for profiling using linux-perf tools without the overhead, although i'm unsure

**Comment by olorin** [comment]
@jystic possibly; it does seem like they could be used independently, though profiling with just `perf-events` is a lot harder than `perf-events` combined with GHC instrumentation. Do you think a separate flag would be a good idea?

**Comment by jacobstanley** [comment]
@olorin yeah, I was thinking something like that, although, maybe we should just keep it simple and wait until we actually have a real need for it before we add another flag.

Let's merge this as is I think :cake:

**Comment by jacobstanley** [comment]
Not sure if @markhibberd has any thoughts on stripping executables?

File: src/Mafia/Init.hs [hunk 1]
```diff
@@ -115,8 +115,15 @@ profilingFlavour = \case
...
-  DisableProfiling -> ["--disable-profiling"]
-  EnableProfiling  -> ["--enable-profiling", "--ghc-options=-fprof-auto-top"]
+  DisableProfiling -> [
+      "--disable-profiling"
+    , "--enable-executable-stripping"
+    ]
+  EnableProfiling  -> [
+      "--enable-profiling"
+    , "--disable-executable-stripping"
+    , "--ghc-options=\"-fprof-auto-top -eventlog\""
```
Comment (jacobstanley) [inline src/Mafia/Init.hs]: Looks like this needs to be:

``` haskell
, "--ghc-options=-fprof-auto-top"
, "--ghc-options=-eventlog"
```

Comment (olorin) [inline src/Mafia/Init.hs]: Yup, fixed.

**Pull request merged**